src/components/modules/document/common/Upload/Sides/BackSide/BackSideController.js

```diff
@@ -182,6 +186,20 @@ class BackSideController extends VuexModule {
       timer.fillAndStopProgress();
     }
   }
+
+  /**
+   * @return {Promise<void>}
+   */
+  @Action
+  async sendFingerprint() {
```

Comment:
- remove this method from both controllers (`BackSideController` and `FrontSideOnlyController`), because they should have context about upload actions only
- create store `riskScoringStore`
- then connect it in `BackSide.vue` and `FrontSideOnly.vue` files this `riskScoringStore` new store
- then in `handleConfirm` method (each component have this method) call `riskScoringStore.sendFingerprint()`

in this solution you don't need copypaste creating of Fingerprint instance.

Reply:
@dudiq done. But after tests noticed that in this solution fingerprint sending after document was uploaded and recognized, after all timeouts. So potentially user can upload document and break up recognizing process, so we got uploaded document without fingerprint. So I guess, we should send it earlier, just after physical file upload, or, maybe, BEFORE upload.

@kazurus what you think?

Reply:
you can put `riskScoringStore.sendFingerprint()` 
for `BackSide.vue` after `await this.$options.backSideController.startUpload({` line
for `FrontSideOnly.vue` after `this.documentId = await this.$options.frontSideController.startCreateDocument(` line
if need create scoring immediatelly after upload

Reply:
yes, if it will be required, it will be done

@kazurus

Reply:
Good solution, please add

Reply:
@dudiq @kazurus done

Reply:
@ifiri call `sendFingerprint` after upload, not before

Reply:
then user can upload document and immediately break next request
anyway, done

Reply:
if something happend with upload, send will be called.
i think, that fingerprint should called with `success` upload, others can be skipped
or what is the reason send data for failed/canceled upload?

Reply:
A document uploading has priority and after that we send the metric

Reply:
done then